ru/basecall.py

```diff
@@ -193,3 +191,30 @@ def map_reads_2(self, calls):
         """
         for read_info, read_id, seq, seq_len, quality in calls:
             yield read_info, read_id, seq_len, list(self.mapper.map(seq))
+
+
+class MappyRSMapper:
+    def __init__(self, index: Union[Path, str]) -> None:
+        self.index = index
+        if self.index:
+            self.mapper = mp.Aligner(6, self.index)
+            self.initialised = True
+        else:
+            self.mapper = None
+            self.initialised = False
+
+    def map_batch(self, iterable: Iterable[tuple[tuple[int, str], dict[Any]]]):
```

Comment:
Not sure that this function definition is what we send. We use (at the moment) read_info as `(channel, read_number)` so should this be `Iterable[tuple[tuple[int, int], ...]]`?

Reply:
I believe this is no correct with [int, int]. I think the channel number and read number are the right way guess I'll find out!